main.go

```diff
@@ -570,12 +594,20 @@ func ProcessDirEntry(s *State, word string, resultChan chan<- Result) {
 }
 
 func PrintDnsResult(s *State, r *Result) {
+	output := ""
 	if r.Status == 404 {
-		fmt.Printf("Missing: %s\n", r.Entity)
+		output += fmt.Sprintf("Missing: %s\n", r.Entity)
 	} else if s.ShowIPs {
-		fmt.Printf("Found: %s [%s]\n", r.Entity, r.Extra)
+		output += fmt.Sprintf("Found: %s [%s]\n", r.Entity, r.Extra)
+	} else if s.ShowCNAME {
+		output += fmt.Sprintf("Found: %s [%s]\n", r.Entity, r.Extra)
 	} else {
-		fmt.Printf("Found: %s\n", r.Entity)
+		output += fmt.Sprintf("Found: %s\n", r.Entity)
+	}
+	fmt.Printf(output)
```

Comment:
OK so to be clear: I don't want to replace the `Sprintf`. If the newlines are already baked into the string, then instead of using `Println` we can use `fmt.Printf("%s", output)`, but we're not going to leave it as `fmt.Printf(output)` :)